src/rhsmlib/services/register.py

```diff
@@ -132,6 +132,19 @@ def register(
             cp_provider = inj.require(inj.CP_PROVIDER)
             cp_provider.close_all_connections()
 
+        access_mode: str = consumer.get("owner", {}).get("contentAccessMode", "unknown")
+        if access_mode != "org_environment":
+            log.error(
+                f"Organization's content access mode is '{access_mode}'. "
+                "Only simple content access/'org_environment' is allowed."
+            )
+            raise exceptions.ServiceError(
+                _(
+                    "Registration is only possible when the organization "
+                    "is in Simple Content Access (SCA) mode."
+                )
```

Comment:
(I thought I had written that, must had lost it...) One important thing to do here is to actually unregister this consumer right before raising the failure, otherwise there will be a "stale" consumer in Candlepin that can consume subscriptions.